knative-operator/pkg/webhook/knativekafka/webhook_validating_test.go

```diff
@@ -185,9 +188,10 @@ func TestInvalidShape(t *testing.T) {
 	os.Clearenv()
 	os.Setenv("REQUIRED_KAFKA_NAMESPACE", "knative-eventing")
 
-	validator := Validator{}
-	validator.InjectDecoder(decoder)
-	validator.InjectClient(fake.NewClientBuilder().WithObjects(validKnativeEventingCR).Build())
+	validator := Validator{
```

Comment:
What's about a `NewValidator()` method as this seems to match the overall scheme better and you probably have to make the members public if added as arguments to that method ?

Reply:
Refactored.